### PR TITLE
refactor: use common multi-tenancy configuration in Zeebe

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Description

Same as it was done for authorizations configuration https://github.com/camunda/camunda/issues/25373

The goal of this pull request is to replace `zeebe.gateway.multiTenancy.enabled` by a common `camunda.security.multiTenancy.enabled`

This should allow to enable multi-tenancy with a single property that is used in the gateway + operate + tasklist (+identity?)

the use of this property in operate and tasklist will be done in a separate PR.

For backward compatibility, `camunda.security.multiTenancy.enabled` defaults to `zeebe.gateway.multiTenancy.enabled` if set. This is applicable to Standalone gateway and Standalone broker (with embedded gateway)

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

relates to https://github.com/camunda/camunda/issues/20455

### CODE REVIEW

File: zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/impl/configuration/GatewayCfgTest.java [hunk 1]
```diff
@@ -30,6 +31,7 @@ public final class GatewayCfgTest {
   private static final String EMPTY_CFG_FILENAME = "/configuration/gateway.empty.yaml";
   private static final String CUSTOM_CFG_FILENAME = "/configuration/gateway.custom.yaml";
   private static final GatewayCfg CUSTOM_CFG = new GatewayCfg();
+  private static final SecurityConfiguration SECURITY_CONFIGURATION = new SecurityConfiguration();
```
Comment (emulbreh) [inline zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/impl/configuration/GatewayCfgTest.java:34]: Is this used?

File: dist/src/main/java/io/camunda/application/commons/configuration/GatewayBasedConfiguration.java [hunk 2]
```diff
@@ -87,11 +86,6 @@ public BrokerClientTimeoutConfiguration brokerClientConfig() {
     return new BrokerClientTimeoutConfiguration(properties.getCluster().getRequestTimeout());
   }
 
-  @Bean
-  public MultiTenancyCfg multiTenancyCfg() {
```
Comment (emulbreh) [inline dist/src/main/java/io/camunda/application/commons/configuration/GatewayBasedConfiguration.java:91]: Can we remove this class or is still used somewhere?

Comment (houssain-barouni) [reply dist/src/main/java/io/camunda/application/commons/configuration/GatewayBasedConfiguration.java:91]: yes I was planning to remove it, then I forgot :/

Comment (emulbreh) [reply dist/src/main/java/io/camunda/application/commons/configuration/GatewayBasedConfiguration.java:91]: I can answer myself: it's still used for backwards compatibility.

Comment (houssain-barouni) [reply dist/src/main/java/io/camunda/application/commons/configuration/GatewayBasedConfiguration.java:91]: No it is not need, we can remove it
the backward compatibility is ensured by
https://github.com/camunda/camunda/blob/f8a4f495cf456d140a6983b197a5e2baf046e484/dist/src/main/resources/application-broker.properties#L22
and
https://github.com/camunda/camunda/blob/f8a4f495cf456d140a6983b197a5e2baf046e484/dist/src/main/resources/application-gateway.properties#L17

File: zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/impl/configuration/GatewayCfgTest.java (hunk 1 above)
Comment (houssain-barouni) [reply zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/impl/configuration/GatewayCfgTest.java:34]: no, I will remove it

File: dist/src/main/java/io/camunda/application/commons/configuration/GatewayBasedConfiguration.java (hunk 2 above)
Comment (emulbreh) [reply dist/src/main/java/io/camunda/application/commons/configuration/GatewayBasedConfiguration.java:91]: There are many references to various *_MULTITENANCY_ENABLED env vars. Will these continue to work or do we need to update them all? For example, there are references to ZEEBE_BROKER_GATEWAY_MULTITENANCY_ENABLED, ZEEBE_GATEWAY_MULTITENANCY_ENABLED and CAMUNDA_OPERATE_MULTITENANCY_ENABLED.

Comment (houssain-barouni) [reply dist/src/main/java/io/camunda/application/commons/configuration/GatewayBasedConfiguration.java:91]: no need to update any parameters used by the standalone applications: if you set 
- `ZEEBE_BROKER_GATEWAY_MULTITENANCY_ENABLED` in a standalone broker, `CAMUNDA_SECRITY_MULTITENANCY_ENABLED` will be set
- `ZEEBE_GATEWAY_MULTITENANCY_ENABLED` in a standalone gateway, `CAMUNDA_SECRITY_MULTITENANCY_ENABLED` will be set
- `CAMUNDA_OPERATE_MULTITENANCY_ENABLED` in a standalone gateway, `CAMUNDA_SECRITY_MULTITENANCY_ENABLED` will be set

**Review by emulbreh — APPROVED** [review]

**Review by koevskinikola — APPROVED** [review]
LGTM @houssain-barouni 👍 

I only have a small nitpick, but feel free to ignore & merge 🙂

File: security/security-core/src/main/java/io/camunda/security/configuration/SecurityConfiguration.java [hunk 3]
```diff
@@ -10,14 +10,25 @@
 public class SecurityConfiguration {
 
   static final boolean DEFAULT_AUTHORIZATIONS_ENABLED = false;
+  static final boolean DEFAULT_MULTITENANCY_ENABLED = false;
```
[the remark this answers is not in the document]
Comment (houssain-barouni) [reply security/security-core/src/main/java/io/camunda/security/configuration/SecurityConfiguration.java:13]: That's a good point, I remember I did similarly to a nested configuration we had somewhere else, but I don't remember which one
I will update it as you are proposing

**Pull request merged**